src/Kiota.Builder/EqualityComparers/GlobComparer.cs

```diff
@@ -9,11 +9,17 @@ internal class GlobComparer : IEqualityComparer<Glob>
 {
     public bool Equals(Glob? x, Glob? y)
     {
-        return x == null && y == null || x != null && y != null && GetHashCode(x) == GetHashCode(y);
+        if (x is not null && y is not null)
+        {
+            return x.ToString().Equals(y.ToString(), StringComparison.OrdinalIgnoreCase);
+        }
+        return x?.Equals(y) == true;
     }
 
     public int GetHashCode([DisallowNull] Glob obj)
     {
-        return obj.ToString().GetHashCode(StringComparison.OrdinalIgnoreCase);
+        var hash = new HashCode();
```

Comment:
I'm not sure this change is an improvement. When we have a single property (previous implementation), we don't need to allocate the additional HashCode.

Reply:
HashCode is a struct. It doesn't allocate any memory.

Reply:
I'm guessing you meant any _heap_ memory? It'll still allocate on the stack.
http://clarkkromenaker.com/post/csharp-structs/

Reply:
Yes. A stack allocation is equivalent to an add instruction. It only increments the stack pointer. I can remove the hash code new if you feel strongly about this one.

Reply:
no strong opinion here. I do understand the value of aligning implementations. If we're down to counting stack allocs, this new implementation allocates 2 more times than the previous implementation though. But this is not a hot path.